include/revng/Pipeline/Context.h

```diff
@@ -120,11 +123,47 @@ class Context {
 
 public:
   llvm::Error storeToDisk(llvm::StringRef Path) const {
-    return Globals.storeToDisk(Path);
+    if (auto Error = Globals.storeToDisk(Path); Error)
+      return Error;
+
+    return llvm::Error::success();
   }
+
   llvm::Error loadFromDisk(llvm::StringRef Path) {
-    return Globals.loadFromDisk(Path);
+    if (auto Error = Globals.loadFromDisk(*this, Path); Error)
+      return Error;
+
+    return llvm::Error::success();
+  }
+
+public:
+  void getTargetsAssociatedToPaths(const TupleTreePath &Path,
+                                   InvalidationMap &Out) const {
+    for (auto &Name : Globals.getGlobalNames()) {
+      llvm::cantFail(Globals.get(Name))->getTargetsAssociatedToPaths(Path, Out);
```

Comment:
`llvm::cantFail(Globals.get(Name))` is weird. Can't we have `GlobalsMap` expose the list of globals without going through names?